core/include/drivers/wdt.h

```diff
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2019 Broadcom.
+ */
+
+#ifndef DRIVERS_WDT_H
+#define DRIVERS_WDT_H
+
+#include <kernel/interrupt.h>
+#include <tee_api_types.h>
+
+struct wdt_chip {
+	const struct wdt_ops *ops;
+	struct itr_handler wdt_itr;
+};
+
+/*
+ * struct wdt_ops - The watchdog device operations
+ *
+ * @start:	The routine for starting the watchdog device.
+ * @stop:	The routine for stopping the watchdog device.
+ * @ping:	The routine that sends a keepalive ping to the watchdog device.
+ * @set_timeout:The routine that finds the load value that will reset system in
+ * required timeout (in seconds).
+ *
+ * The wdt_ops structure contains a list of low-level operations
+ * that control a watchdog device.
+ */
+struct wdt_ops {
+	void (*start)(struct wdt_chip *chip);
+	void (*stop)(struct wdt_chip *chip);
+	void (*ping)(struct wdt_chip *chip);
+	TEE_Result (*set_timeout)(struct wdt_chip *chip, unsigned long timeout);
+};
+
```

Comment:
Since field `ops` in the structure, maybe this header could define few inlines:
```C
static inline void wdt_start(struct wdt_chip *chip)
{
	assert(chip->start);
	chip->start(chip);
}

static inline void wdt_stop(struct wdt_chip *chip)
{
	assert(chip->start);
	chip->stop(chip);
}

...
```

Reply:
wdt_start, wdt_end, etc. should be reserved for when wdt framework get added, similar to console_putc, console_flush in the existing console framework?

Reply:
I just thought that for one to call wdt start/stop/ping, one would need to call through a `struct wdt_chip` reference, something like `ref->start(ref)` which would be more clear implemented as `wdt_start(ref);`.

Actually this is not an issue in your P-R. Could be implemented later if really useful.